### ISSUE TITLE
[question] Can I cross-compile qt 5.15.2 for the Raspberry Pi?

### ISSUE THREAD

**Issue by agn0stico** [conan-io/conan-center-index#5708]
Hey guys,

did anyone ever try to cross compile qt/5.15.2 for the Raspberry Pi 4 host (armv8)?

I tried to use [this](https://www.interelectronix.com/qt-515-cross-compilation-raspberry-compute-module-4-ubuntu-20-lts.html) as a guide but I noticed that the conan recipe does not allow setting some required options (for example, the _eglfs_ switch).

What is your experience? Thanks!

**Comment by ericLemanissier** [comment]
I did not try to cross compile, but you can use the `config` option to set arbitrary command line switches, like this: `-o qt:config="-eglfs"`

**Comment by agn0stico** [comment]
Thanks @ericLemanissier , that suggestion actually got me a step further. Unfortunately it seems something is going wrong during configuration. At the end of the configuration phase I get some errors:

```
ERROR: Feature 'zstd' was enabled, but the pre-condition 'libs.zstd' failed.

ERROR: Feature 'opengles2' was enabled, but the pre-condition '(config.win32 && !features.opengl-dynamic) || (!config.watchos && !features.opengl-desktop && libs.opengl_es2)' failed.

ERROR: Feature 'eglfs' was enabled, but the pre-condition '!config.android && !config.darwin && !config.win32 && !config.wasm && features.egl' failed.

ERROR: Feature 'system-freetype' was enabled, but the pre-condition 'features.freetype && libs.freetype' failed.

ERROR: The OpenGL functionality tests failed!
You might need to modify the include and library search paths by editing QMAKE_INCDIR_OPENGL[_ES2],
QMAKE_LIBDIR_OPENGL[_ES2] and QMAKE_LIBS_OPENGL[_ES2] in the mkspec for your platform.
```

I might be messing up something with the dependencies or the sysroot.

I am trying to cross-build it from within an Ubuntu 20.04 virtual machine. I am using the toolchain `gcc-linaro-7.5.0-2019.12-rc1-x86_64_armv8l-linux-gnueabihf` which I know works fine because I successfully cross compiled some other applications.

My host profile looks like this:

```
target_host=armv8l-linux-gnueabihf
standalone_toolchain=/home/pi/Desktop/xbuild/gcc-linaro-7.5.0-2019.12-rc1-x86_64_armv8l-linux-gnueabihf
cc_compiler=gcc
cxx_compiler=g++

[settings]
os=Linux
arch=armv8
compiler=gcc
compiler.version=7.5
compiler.libcxx=libstdc++11
compiler.cppstd=gnu14
build_type=Release

[env]
CONAN_CMAKE_FIND_ROOT_PATH=$standalone_toolchain/$target_host
CONAN_CMAKE_SYSROOT=/home/pi/Desktop/xbuild/sysroot
PATH=[$standalone_toolchain/bin]
CHOST=$target_host
AR=$target_host-ar
AS=$target_host-as
RANLIB=$target_host-ranlib
LD=$target_host-ld
STRIP=$target_host-strip
CC=$target_host-$cc_compiler
CXX=$target_host-$cxx_compiler
CXXFLAGS=-I"$standalone_toolchain/$target_host/include"

[options]
qt:config="-eglfs"
qt:with_icu=False
qt:with_fontconfig=False
qt:opengl=es2
qt:device=linux-rasp-pi4-v3d-g++
qt:cross_compile=$standalone_toolchain/bin/armv8l-linux-gnueabihf-
qt:sysroot=/home/pi/Desktop/xbuild/sysroot
```

and my build profile is:

```
[settings]
os=Linux
arch=x86_64
compiler=gcc
compiler.version=9
compiler.libcxx=libstdc++11
compiler.cppstd=gnu17
build_type=Release
```

I am also configuring qt in my conanfile as following:

```
self.options["qt"].shared = True
self.options["qt"].with_harfbuzz = False
self.options["qt"].openssl = False
self.options["qt"].with_sqlite3 = False
self.options["qt"].with_mysql = False
self.options["qt"].with_pq = False
self.options["qt"].with_odbc = False
self.options["qt"].with_libjpeg = "libjpeg"
```

This is the resulting [config.log](https://github.com/conan-io/conan-center-index/files/6634653/config.log). Does anyone have any suggestion where to look for issues? Or would you rather suggest to write on the Qt Forum?

**Comment by cyreon** [comment]
Have you managed to resolve this issue? I tried to cross-compile to Raspberry Pi 5 target, but build fails. I spent total 3 days to make Qt6 work with Raspberry Pi, but in the end, to no avail. I'm seriously thinking just ditching Conan with Raspberry and use something else.

**Comment by matte86** [comment]
Unfortunately, I never got this to work and then moved to a different job, so I kind of dropped this. I have not been using Conan ever since, so I won't be able to comment on any progress. I only recall some back-and-forth discussion.s on the Qt forum, but I cannot remember all the details and don't have any link to the discussions, unfortunately.

**Comment by cyreon** [comment]
Yeah, this topic is really a mess, unfortunately. I'm doing the same: asking questions, pushing claude and GPT to do research for me: no avail. I feel like an idiot to be honest. The biggest achievement so far was to compile QML project with QSerialPort successfully, but in the end, Qt app failed during runtime due missing plugins or libraries even though I had them in environment variables and double checked everything at least 10 times.

Since it's just an uni project, I'll either just install stuff with APT or ditch Qt for WxWidgets or something else altogether.

Actually I need a proper Serial I/O interface with proper asynchronous abilities and some widget system I can easily use to simulate car dashboard.
